### PR TITLE
feat: add type declaration for Structure, History and ProjectManager

### PR DESCRIPTION
**Pull request by Pduhard**
add type declaration for Structure, History and ProjectManager

### CODE REVIEW

File: types/lib/history.d.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
+import Class from "abitbol";
+
+export class History extends Class {
+
+    __init__(pm: any, params: any): void;
```
Comment (jbghoul-wanadev) [inline types/lib/history.d.ts:5]: pm: ProjectManager

Comment (jbghoul-wanadev) [reply types/lib/history.d.ts:5]:
```ts
params?: { maxLength?: number }
```

File: types/lib/project-manager.d.ts [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+    type: string
+    metadataFormat: string
+    projectFormat: string
+    blobIndexFormat: string
+}
+
+export class ProjectManager extends SerializableClass {
+    __name__: "Project";
+
+    __init__(params: unknown): void;
```
Comment (jbghoul-wanadev) [inline types/lib/project-manager.d.ts:14]: params: {
version,
mimetype,
fileExt
...
}

there are only 5 fields to define

File: types/lib/project-manager.d.ts [hunk 3]
```diff
@@ -0,0 +1,294 @@
+import SerializableClass from "abitbol-serializable";
+import type { Structure } from "./structure";
+
+interface WprjOptions {
```
Comment (jbghoul-wanadev) [inline types/lib/project-manager.d.ts:4]: could be exported

**Review by jbghoul-wanadev — CHANGES_REQUESTED** [review]

File: types/lib/project-manager.d.ts (hunk 3 above)
Comment (Pduhard) [reply types/lib/project-manager.d.ts]: https://github.com/wanadev/obsidian-project/pull/55/commits/81b8ff3b918a06b9ccda53c966e92166ad6502f3

**Review by jbghoul-wanadev — APPROVED** [review]

**Pull request merged**